crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/AppealContributionService.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.laa.crime.contribution.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.contribution.builder.AppealContributionResponseBuilder;
+import uk.gov.justice.laa.crime.contribution.builder.CreateContributionRequestBuilder;
+import uk.gov.justice.laa.crime.contribution.builder.GetContributionAmountRequestBuilder;
+import uk.gov.justice.laa.crime.contribution.model.*;
+import uk.gov.justice.laa.crime.contribution.staticdata.enums.AssessmentResult;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AppealContributionService {
+
+    private final MaatCourtDataService maatCourtDataService;
+    private final GetContributionAmountRequestBuilder getContributionAmountRequestBuilder;
+    private final CreateContributionRequestBuilder createContributionRequestBuilder;
+    private final AppealContributionResponseBuilder appealContributionResponseBuilder;
+
+    public AppealContributionResponse calculateContribution(AppealContributionRequest appealContributionRequest, String laaTransactionId) {
+        AssessmentResult assessmentResult = determineAssessmentResult(appealContributionRequest.getAssessments());
+
+        GetContributionAmountRequest getContributionAmountRequest = getContributionAmountRequestBuilder.build(appealContributionRequest, assessmentResult);
+        BigDecimal appealContributionAmount = maatCourtDataService.getContributionAppealAmount(getContributionAmountRequest, laaTransactionId);
```

Comment:
Maybe try and simplify this section a little by extracting another private method i.e.

        BigDecimal appealContributionAmount = determineAppealContributionAmount(appealContributionRequest, laaTransactionId);

...

    private BigDecimal determineAppealContributionAmount(AppealContributionRequest appealContributionRequest, String laaTransactionId) {
        AssessmentResult assessmentResult = determineAssessmentResult(appealContributionRequest.getAssessments());

        GetContributionAmountRequest getContributionAmountRequest = getContributionAmountRequestBuilder.build(appealContributionRequest, assessmentResult);
        return maatCourtDataService.getContributionAppealAmount(getContributionAmountRequest, laaTransactionId);
    }